src/main/java/seedu/address/logic/commands/AddAttendanceCommand.java

```diff
@@ -0,0 +1,166 @@
+ */
+public class AddAttendanceCommand extends Command {
+
+    public static final String COMMAND_WORD = "add-attendance";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Adds a student's attendance to the specified "
+            + "lesson in the student manager. "
+            + "Note: All indexes must be a positive integer.\n"
+            + "Parameters: "
+            + PREFIX_CLASS_INDEX + "CLASS_INDEX "
+            + PREFIX_LESSON_INDEX + "LESSON_INDEX "
+            + PREFIX_STUDENT_INDEX + "STUDENT_INDEX "
+            + PREFIX_WEEK + "WEEK_NUMBER (must be positive integer) "
+            + PREFIX_PARTICIPATION_SCORE + "PARTICIPATION_SCORE (must be an integer between 0 and 100)";
+
+    public static final String MESSAGE_SUCCESS = "New attendance added: %1$s attended week %2$s lesson with "
+            + "participation score of %3$s";
+    public static final String MESSAGE_DUPLICATE_ATTENDANCE = "Attendance have been recorded previously.";
+
+    private final Index moduleCLassIndex;
+    private final Index lessonIndex;
+    private final Index studentIndex;
+    private final Week week;
+    private final Attendance toAdd;
+
+    /**
+     * Creates an AddAttendanceCommand to add the specified {@code Attendance}.
+     */
+    public AddAttendanceCommand(Index moduleCLassIndex, Index lessonIndex, Index studentIndex,
+                                Week week, Attendance toAdd) {
+        requireAllNonNull(moduleCLassIndex, lessonIndex, studentIndex, week, toAdd);
+
+        this.moduleCLassIndex = moduleCLassIndex;
+        this.lessonIndex = lessonIndex;
+        this.studentIndex = studentIndex;
+        this.week = week;
+        this.toAdd = toAdd;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+
+        List<Student> lastShownStudentList = model.getFilteredStudentList();
+        List<ModuleClass> lastShownModuleClassList = model.getFilteredModuleClassList();
+
+        if (studentIndex.getZeroBased() >= lastShownStudentList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_STUDENT_DISPLAYED_INDEX);
+        }
+
+        if (moduleCLassIndex.getZeroBased() >= lastShownModuleClassList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_MODULE_CLASS_DISPLAYED_INDEX);
+        }
+
+        ModuleClass targetModuleClass = lastShownModuleClassList.get(moduleCLassIndex.getZeroBased());
+        Student targetStudent = lastShownStudentList.get(studentIndex.getZeroBased());
+
+        if (lessonIndex.getZeroBased() >= targetModuleClass.getLessons().size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_LESSON_DISPLAYED_INDEX);
+        }
+
+        Lesson targetLesson = targetModuleClass.getLessons().get(lessonIndex.getZeroBased());
+
+        if (!targetLesson.getAttendanceRecordList().isWeekContained(week)) {
```

Comment:
We need to perform some checks to ensure that the student is a member of the `moduleClass` as well.